dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/appmanager/DefaultAppManager.java

```diff
@@ -273,35 +273,41 @@ public App getApp(String key, String contextPath) {
   }
 
   @Override
-  public AppStatus installApp(File file, String fileName) {
+  public App installApp(File file, String fileName) {
     App app = jCloudsAppStorageService.installApp(file, fileName, appCache);
 
+    log.info(
+        String.format(
+            "Installed App with ID %s (status: %s)", app.getAppHubId(), app.getAppState()));
+
     if (app.getAppState().ok()) {
       appCache.put(app.getKey(), app);
       registerDatastoreProtection(app);
     }
 
-    return app.getAppState();
+    return app;
   }
 
   @Override
-  public AppStatus installApp(UUID appHubId) {
+  public App installApp(UUID appHubId) {
+
+    App installedApp = new App();
     if (appHubId == null) {
-      return AppStatus.NOT_FOUND;
+      installedApp.setAppState(AppStatus.NOT_FOUND);
     }
 
     try {
       String versionJson = appHubService.getAppHubApiResponse("v2", "appVersions/" + appHubId);
```

Comment:
I think this will send a request to `appVersions/null` if the appHubId is not found.  We need to keep the early-exit we had with the `return` above